tests/gtest/CMakeLists.txt

```diff
@@ -1,4 +1,14 @@
-set(GTEST_ROOT_DIR ${TR_THIRD_PARTY_SOURCE_DIR}/googletest/googletest)
+find_path(GTEST_ROOT_DIR
+    NAMES
+        src/gtest-all.cc
```

Comment:
Are you testing this locally? I'm not sure I know of a system that provides this file via an installable package so looking for it is most probably wrong. I'd say we should just try to `find_package(GTest)` first, and then fallback to defining the `GTest::gtest_main` target ourselves if not found.

Same goes for other packages if they support installation and export CMake target(s).

Reply:
Thanks for the feedback.
I am of course testing locally and it builds fine (at least on Linux/amd64).
GTest source files are provided at least on Debian by https://packages.debian.org/googletest and we're already using this for building.
I'll rework this to move from `find_path` to `find_package` when possible (fast_float, fmt and utfcpp all provide `cmake` package files so this seems to be possible) and fix the CI issue on android and Gentoo/s390 (absolutely no clue about that, will look into a `cmake` version mismatch and behaviour change).

Reply:
I have no idea why Debian does that, seems counterintuitive — it's not a source package. Gentoo doesn't: https://www.portagefilelist.de/index.php?p=package&id=e82837aede7bf0b74e3535ff15eaef44